### DISCUSSION TITLE
License of generated facades

### DISCUSSION THREAD

**Discussion by mpilquist** [ScalablyTyped/Converter#397] [General]
The converter is licensed under GPL v3 whereas the rest of ScalablyTyped is licensed under MIT. Does the GPLv3 license apply only to the Converter itself and not facades generated by the converter? If so, could that be clarified somewhere in docs? Thanks!

**Comment by oyvindberg** [comment]
Hey there @mpilquist 

Yes, that's the gist of it. 

I chose the GPL3 license for the Converter itself because I think it's a good license for open source programs. 

However, I do not want the viral nature of it to affect projects using it to generate code. As such:
- the accompanying [runtime artifact](https://github.com/ScalablyTyped/Runtime) is MIT-licensed.
- no usage constraints are intended to apply to the generated facades. I have said in the past that they are MIT licensed as well, but I'm not sure if that even makes sense to say. I suppose they may be considered derived work of the input typescript files, and may fall under the corresponding license. Or if they are not, whoever publishes a facade is free to choose a suitable license.

I should add that the GPL3 license was chosen before the sbt plugin was written, but I don't think/hope it complicates the picture. If you have any further concerns about the current situation I'd be happy to discuss changes. Otherwise if you think this is fine I'll go ahead and add it to the web page.

**Comment by mpilquist** [comment]
Makes sense to me. Thanks very much for clarifying!
